### PR TITLE
Linear regression using straight line@abhishekgupta368

### PR DESCRIPTION
**Pull request by abhishekgupta368**
Linear regression equation using straight line #1651
Fixes #

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
-  Added and calculated Linear Regression using Straight line
#### Languages Used:
- python
#### Files Added:
- Added_Linear_Regression_using_Straight_line.py
> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by asha15** [comment]
@abhisekssp4025 you have added the `.cpp` file too. please remove it and let me know

**Comment by abhishekgupta368** [comment]
ok

**Comment by abhishekgupta368** [comment]
done

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        x=0
+        y=0
+        sqr_x=0
+        for i in range(n):
+            xy = xy + self.__arr_y[i]*self.__arr_x[i]
+            x = x + self.__arr_x[i]
+            y = y + self.__arr_y[i]
+            sqr_x = sqr_x + self.__arr_x[i]*self.__arr_x[i]
+        
+        a0 = (y*sqr_x - x*xy)/(self.__n*sqr_x - x*x)
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: keep space before and after operators

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        return a0
+    """
+        return the value of equation
+        which is 
+        y = mx+c
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: take this to the previous line

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    """
+    def prediction(self,x):
+        y = self.slope() + self.intercept()*x
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: keep space after comma
keep space before and after operators

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        return y
+    
+if __name__ == "__main__":
+#    n = 7
+#    x = [1,2,3,4,5,6,7]
+#    y = [0.5,2.5,2.0,4.0,3.5,6.0,5.5]
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: Is this are sample inputs ?

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    n = int(input("Enter size: "))
+    arr_x = list(map(int,input("Enter x array: ").strip().split(',')))
+    arr_y = list(map(float,input("Enter y array: ").strip().split(',')))
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: keep space after comma

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        print("Calculated slope: ",ls.slope())
+        
+        while(True):
+            flag = str(input("Do you want to predict values(y/n): "))
+            if(flag=="n"):
+                print("System Exit")
+                break
+            elif(flag=="y"):
+                c = int(input("Enter value for prediction: "))
+                print("Predicted value: ",ls.prediction(c))
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: keep space after comma
keep space before and after operators

**Comment by abhishekgupta368** [comment]
done

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
+class LinearRegression():
+    """Initialize arr_x and arr_y"""
+    def __init__(self , arr_x , arr_y , n):
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: Keep a space after the comma, not before

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        y = 0
+        sqr_x = 0
+        for i in range(n):
+            xy = xy + self.__arr_y[i] * self.__arr_x[i]
+            x = x + self.__arr_x[i]
+            y = y + self.__arr_y[i]
+            sqr_x = sqr_x + self.__arr_x[i] * self.__arr_x[i]
+        
+        a1 = (self.__n * xy - x*y) / (self.__n * sqr_x - x * x)
+        return a1
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: keep a line space after this

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    """calculate the the slope of line"""
+    def slope(self):
+        xy = 0
+        x = 0
+        y = 0
+        sqr_x = 0
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: keep a line space after this

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+        for i in range(n):
+            xy = xy + self.__arr_y[i] * self.__arr_x[i]
+            x = x + self.__arr_x[i]
+            y = y + self.__arr_y[i]
+            sqr_x = sqr_x + self.__arr_x[i] * self.__arr_x[i]
+        
+        a0 = (y * sqr_x - x * xy) / (self.__n * sqr_x - x * x)
+        return a0
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: keep a line space after this

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+    """return the value using line of equation: y = m * x + c"""
+    def prediction(self , x):
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: Remove the space before comma

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+        y = self.slope() + self.intercept() * x
+        return y
+    
+if __name__ == "__main__":
+    n = int(input("Enter size: "))
+    arr_x = list(map(int , input("Enter x array: ").strip().split(',')))
+    arr_y = list(map(float , input("Enter y array: ").strip().split(',')))
+    
+    if(n==len(arr_x) and n==len(arr_y)):
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: keep space before and after ==

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+        ls = LinearRegression(arr_x , arr_y , n)
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: remove the space before comma

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+        return a0
+    """return the value using line of equation: y = m * x + c"""
+    def prediction(self , x):
+        y = self.slope() + self.intercept() * x
+        return y
+    
+if __name__ == "__main__":
+    n = int(input("Enter size: "))
+    arr_x = list(map(int , input("Enter x array: ").strip().split(',')))
+    arr_y = list(map(float , input("Enter y array: ").strip().split(',')))
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: remove the space before comma

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+    
+    if(n==len(arr_x) and n==len(arr_y)):
+        ls = LinearRegression(arr_x , arr_y , n)
+        print("Calculated intercept: ", ls.intercept())
+        print("Calculated slope: ", ls.slope())
+        
+        while(True):
+            flag = str(input("Do you want to predict values(y/n): "))
+            if(flag=="n"):
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: keep space before and after ==

File: Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+                print("System Exit")
+                break
+            elif(flag=="y"):
```
Comment (asha15) [inline Linear_Regression_Using_Straight_Line/Linear_Regression_using_Straight_Line.py]: keep space before and after ==

**Comment by asha15** [comment]
squash your commits and push it

**Comment by abhishekgupta368** [comment]
don

**Comment by abhishekgupta368** [comment]
please somebody, review it

**Pull request merged**